internal/xds/rbac/rbac_engine.go

```diff
@@ -238,3 +272,39 @@ type rpcData struct {
 	// handshake.
 	certs []*x509.Certificate
 }
+
+func (e *engine) doAuditLogging(rpcData *rpcData, rule string, authorized bool) {
+	// In the RBAC world, we need to have a SPIFFE ID as the principal for this
+	// to be meaningful
+	principal := ""
+	if len(rpcData.certs) != 0 && len(rpcData.certs[0].URIs) != 0 &&
+		rpcData.certs[0].URIs[0].Scheme == "spiffe" {
+		principal = rpcData.certs[0].URIs[0].String()
+	}
+
+	event := &audit.Event{
+		FullMethodName: rpcData.fullMethod,
+		Principal:      principal,
+		PolicyName:     e.policyName,
+		MatchedRule:    rule,
+		Authorized:     authorized,
+	}
+	for _, logger := range e.auditLoggers {
+		switch e.auditCondition {
```

Comment:
should there be a case for `RBAC_AuditLoggingOptions_NONE` or maybe a default path for the switch?

Reply:
It's a no-op if it's `NONE` or anything else

Would it be clearer to have no-op cases?

Reply:
I was only wondering. This lgtm for no-op.